tests/test_version_traversal.py

```diff
@@ -308,15 +313,139 @@ def test_relative_upgrade(self):
             set([self.c2.revision]),
         )
 
-    def test_relative_downgrade(self):
+    def test_relative_downgrade_baseplus2(self):
+        """ base+2 points to b, no branch label, drop everything above b. """
+        self._assert_downgrade(
+            "base+2",
+            [self.d2.revision, self.d1.revision],
+            [
+                self.down_(self.d2),
+                self.down_(self.c2),
+                self.down_(self.d1),
+                self.down_(self.c1),
+            ],
+            set([self.b.revision]),
+        )
 
+    def test_relative_downgrade_branchplus2(self):
+        """
+        Correct behaviour (per
+        https://github.com/sqlalchemy/alembic/pull/763#issuecomment-738741297)
+        Only the c2branch should be downgraded, right back to base+2 = b
+        """
         self._assert_downgrade(
             "c2branch@base+2",
             [self.d2.revision, self.d1.revision],
-            [self.down_(self.d2), self.down_(self.c2), self.down_(self.d1)],
+            [
+                self.down_(self.d2),
+                self.down_(self.c2),
+            ],
+            set([self.d1.revision]),
+        )
+
+    def test_relative_downgrade_branchplus3(self):
+        """ c2branch@base+3 equivalent to c2. """
+        self._assert_downgrade(
+            self.c2.revision,
+            [self.d2.revision, self.d1.revision],
+            [self.down_(self.d2)],
+            set([self.d1.revision, self.c2.revision]),
+        )
+        self._assert_downgrade(
+            "c2branch@base+3",
+            [self.d2.revision, self.d1.revision],
+            [self.down_(self.d2)],
+            set([self.d1.revision, self.c2.revision]),
+        )
+
+    # Old downgrade -1 behaviour depends on order of branch upgrades.
+    # This should probably fail (ambiguous) but is currently documented
+    # as a key use case in branching.
+
+    @pytest.mark.xfail
```

Comment:
**mike bayer** (zzzeek) wrote:

we have our own decorators here:

    from alembic.testing import exclusions

    # ...

    @exclusions.fails()
    def test_thing(...):

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/alembic/+/2530

Reply:
**Simon Bowly** (simonbowly) wrote:

Done

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/alembic/+/2530